app/queries/servers/post.ts

```diff
@@ -190,6 +190,14 @@ export const queryPostsById = (database: Database, postIds: string[], sort?: Q.S
     return database.get<PostModel>(POST).query(...clauses);
 };
 
+export const queryPostsByType = (database: Database, type: string, sort?: Q.SortOrder) => {
+    const clauses: Q.Clause[] = [Q.where('type', type)];
```

Comment:
If we want to search by `type` in the `Post` table, we should probably index the column?  And this could probably require migration as well?  Thoughts @larkox @enahum?

Reply:
it would be best to index it yes.. specially true if there are way too many posts stored in the database, although there is a clean up already happening set for every two weeks or so.

Any work that needs to be done at the DB level and it's migration should be on a separate PR and then once merged, we can update this PR base branch.

Reply:
Ah interesting.

If you have a chance, can you tell me where is the clean up code?

Reply:
I think is the function in https://github.com/mattermost/mattermost-mobile/blob/main/app/actions/local/systems.ts#L221